### PR TITLE
PB-16346: Added timezone support to SDK.

### PR DESCRIPTION
**Pull request by marcinsliwinskivasco**

### CODE REVIEW

File: sdk/src/main/java/com/silanis/esl/sdk/builder/PackageBuilder.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
     private SenderInfo senderInfo = null;
     private DocumentPackageAttributes attributes;
     private Visibility visibility;
+    private String timezoneId;
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/builder/PackageBuilder.java]: I can't understand why a package should have a timezone. Is not the time zone necessary for the user? Please link the Acceptance Criteria of the relevant ticket.

File: sdk/src/main/java/com/silanis/esl/sdk/examples/BasicPackageCreationExample.java [hunk 2]
```diff
@@ -47,6 +48,7 @@ public void execute() {
         email2 = "CapitalLetters@email.com";
 
         DocumentPackage superDuperPackage = newPackageNamed(getPackageName())
+                .withTimezoneId(PACKAGE_TIMEZONE_ID)
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/examples/BasicPackageCreationExample.java]: I can't understand why a package should have a timezone. Is not the time zone necessary for the user? Please link the Acceptance Criteria of the relevant ticket.

File: sdk/src/main/java/com/silanis/esl/api/model/BasePackage.java [hunk 3]
```diff
@@ -88,6 +90,7 @@ public BasePackage ( ) {}
     protected java.util.Date _updated;
     protected String _visibility = "ACCOUNT";
     protected java.util.Date _created;
+    protected String _timezoneId;
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/api/model/BasePackage.java]: I can't understand why a package should have a timezone. Is not the time zone necessary for the user? Please link the Acceptance Criteria of the relevant ticket.

File: sdk/src/main/java/com/silanis/esl/sdk/builder/PackageBuilder.java (hunk 1 above)
Comment (marcinsliwinskivasco) [reply sdk/src/main/java/com/silanis/esl/sdk/builder/PackageBuilder.java]: This is one of the major ideas behind the solution we discussed and agreed upon a few months back when we prepared the overall design for timezone support implementation. There was even a whole user story to implement this, please refer to https://jira.vasco.com/browse/PB-12568

Currently the API allows for retrieving/setting the timezone on a package, and that's why we're adding it to the SDK as well.

File: sdk/src/main/java/com/silanis/esl/sdk/examples/BasicPackageCreationExample.java (hunk 2 above)
Comment (marcinsliwinskivasco) [reply sdk/src/main/java/com/silanis/esl/sdk/examples/BasicPackageCreationExample.java]: Please see above comment for explanation.

File: sdk/src/main/java/com/silanis/esl/api/model/BasePackage.java (hunk 3 above)
Comment (marcinsliwinskivasco) [reply sdk/src/main/java/com/silanis/esl/api/model/BasePackage.java]: Please see above comment for explanation.

File: sdk/src/main/java/com/silanis/esl/api/model/Sender.java [hunk 4]
```diff
@@ -457,7 +459,19 @@ public Sender safeSetUpdated( java.util.Date value ){
...
+        super.setTimezoneId(id);
+        return this;
+    }
+
+    @JsonIgnore
+    @Override
+    public Sender safeSetTimezoneId(String id) {
+        if ( id != null ) { this.setTimezoneId(id); }
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (SeungChoi) [reply sdk/src/main/java/com/silanis/esl/api/model/Sender.java]: If so, the backend code might be incorrect. Anyway, this code is not needed in the SDK, so please remove it

File: sdk/src/main/java/com/silanis/esl/api/model/Sender.java [hunk 5]
```diff
@@ -26,6 +26,8 @@
     @JsonIgnore
     public static final String FIELD_COMPANY = "company";
     @JsonIgnore
+    public static final String FIELD_TIMEZONE_ID = "timezoneId";
+    @JsonIgnore
```
[the remark this answers is not in the document]
Comment (SeungChoi) [reply sdk/src/main/java/com/silanis/esl/api/model/Sender.java]: If so, the backend code might be incorrect. Anyway, this code is not needed in the SDK, so please remove it

File: sdk/src/main/java/com/silanis/esl/api/model/Package.java [hunk 6]
```diff
@@ -228,7 +230,19 @@ public Package safeSetLanguage( String value ){
...
+    public Package setTimezoneId( String value ){
+        super.setTimezoneId(value);
+        return this;
+    }
+    // Used internally by aws. Invokes a the corresponding setter if the value is not null
+    @JsonIgnore
+    public Package safeSetTimezoneId( String value ){
+        if ( value != null ) { this.setTimezoneId( value ); }
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (SeungChoi) [reply sdk/src/main/java/com/silanis/esl/api/model/Package.java]: If so, the backend code might be incorrect. Anyway, this code is not needed in the SDK, so please remove it

File: sdk/src/main/java/com/silanis/esl/api/model/Package.java [hunk 7]
```diff
@@ -37,6 +37,8 @@
     @JsonIgnore
     public static final String FIELD_LANGUAGE = "language";
     @JsonIgnore
+    public static final String FIELD_TIMEZONE_ID = "timezoneId";
```
[the remark this answers is not in the document]
Comment (SeungChoi) [reply sdk/src/main/java/com/silanis/esl/api/model/Package.java]: If so, the backend code might be incorrect. Anyway, this code is not needed in the SDK, so please remove it

**Pull request merged**